pyparsing/diagram/__init__.py

```diff
@@ -294,40 +325,64 @@ def _to_diagram_element(
     # Python's id() is used to provide a unique identifier for elements
     el_id = id(element)
 
+    element_results_name = element.resultsName
+    ret = None
+
+    # # some elements don't go in the diagram at all
+    # if isinstance(element, (pyparsing.FollowedBy, pyparsing.PrecededBy)):
+    #     return
+
     # Here we basically bypass processing certain wrapper elements if they contribute nothing to the diagram
-    if isinstance(element, (pyparsing.Group, pyparsing.Forward)) and (
-        not element.customName or not exprs[0].customName
-    ):
-        # However, if this element has a useful custom name, we can pass it on to the child
-        if not exprs[0].customName:
-            propagated_name = name
-        else:
-            propagated_name = None
-
-        return _to_diagram_element(
-            element.expr,
-            parent=parent,
-            lookup=lookup,
-            vertical=vertical,
-            index=index,
-            name_hint=propagated_name,
-        )
+    if not element.customName:
+        if isinstance(element,
+                      (
+                          pyparsing.TokenConverter,
+                          pyparsing.Forward,
+                          pyparsing.Located,
+                      )
+                      ):
+            # However, if this element has a useful custom name, we can pass it on to the child
```

Comment:
"if this element has a useful custom name, and its child does not"

Reply:
Are you suggesting a change here?

Reply:
Ah, okay.

Will do.